code/datums/components/storage/storage.dm

```diff
@@ -233,7 +233,7 @@
 
 /datum/component/storage/proc/quick_empty(mob/M)
 	var/atom/A = parent
-	if((!ishuman(M) && (A.loc != M)) || (M.stat != CONSCIOUS) || M.restrained() || !M.canmove)
+	if(!ishuman(M) || !M.canUseTopic(src, BE_CLOSE = TRUE))

```

Comment:
A) src is datum here
B) This is not proper place to use canUseTopic, this should be abstracted as mob property/function.

Reply:
so like a mob proc that checks if theyre unconscious, restrained, etc?

Reply:
dont we probably have one of those already

Reply:
No i mean that is separate, i mean one that checks if mob is allowed to use storage at all.

Reply:
That parts stays what i want is separate check to see if mob is allowed to use storage at all.